TestFullParquetReader>AbstractTestParquetReader.testDecimalBackedByINT32

Haven't fully traced through the code yet, but this one might be caused by an unfortunate choice of random values in the test.

```
2024-03-06T21:03:24.6089444Z [ERROR] Tests run: 2934, Failures: 1, Errors: 0, Skipped: 90, Time elapsed: 2,531.069 s <<< FAILURE! - in TestSuite
2024-03-06T21:03:24.6091508Z [ERROR] com.facebook.presto.hive.parquet.TestFullParquetReader.testDecimalBackedByINT32  Time elapsed: 5.949 s  <<< FAILURE!
2024-03-06T21:03:24.6093559Z java.lang.IllegalArgumentException: maxCapacityHint can't be less than initialSlabSize 1024 100
2024-03-06T21:03:24.6094940Z 	at org.apache.parquet.Preconditions.checkArgument(Preconditions.java:97)
2024-03-06T21:03:24.6098576Z 	at org.apache.parquet.bytes.CapacityByteArrayOutputStream.<init>(CapacityByteArrayOutputStream.java:153)
2024-03-06T21:03:24.6100792Z 	at org.apache.parquet.column.values.delta.DeltaBinaryPackingValuesWriter.<init>(DeltaBinaryPackingValuesWriter.java:88)
2024-03-06T21:03:24.6103134Z 	at org.apache.parquet.column.values.delta.DeltaBinaryPackingValuesWriterForInteger.<init>(DeltaBinaryPackingValuesWriterForInteger.java:71)
2024-03-06T21:03:24.6105615Z 	at org.apache.parquet.column.values.delta.DeltaBinaryPackingValuesWriterForInteger.<init>(DeltaBinaryPackingValuesWriterForInteger.java:66)
2024-03-06T21:03:24.6108456Z 	at org.apache.parquet.column.values.factory.DefaultV2ValuesWriterFactory.getInt32ValuesWriter(DefaultV2ValuesWriterFactory.java:93)
2024-03-06T21:03:24.6110796Z 	at org.apache.parquet.column.values.factory.DefaultV2ValuesWriterFactory.newValuesWriter(DefaultV2ValuesWriterFactory.java:63)
2024-03-06T21:03:24.6113030Z 	at org.apache.parquet.column.values.factory.DefaultValuesWriterFactory.newValuesWriter(DefaultValuesWriterFactory.java:52)
2024-03-06T21:03:24.6114935Z 	at org.apache.parquet.column.ParquetProperties.newValuesWriter(ParquetProperties.java:167)
2024-03-06T21:03:24.6116395Z 	at org.apache.parquet.column.impl.ColumnWriterBase.<init>(ColumnWriterBase.java:84)
2024-03-06T21:03:24.6117710Z 	at org.apache.parquet.column.impl.ColumnWriterV2.<init>(ColumnWriterV2.java:65)
2024-03-06T21:03:24.6119207Z 	at org.apache.parquet.column.impl.ColumnWriteStoreV2.createColumnWriter(ColumnWriteStoreV2.java:44)
2024-03-06T21:03:24.6120805Z 	at org.apache.parquet.column.impl.ColumnWriteStoreBase.<init>(ColumnWriteStoreBase.java:125)
2024-03-06T21:03:24.6122263Z 	at org.apache.parquet.column.impl.ColumnWriteStoreV2.<init>(ColumnWriteStoreV2.java:38)
2024-03-06T21:03:24.6123782Z 	at org.apache.parquet.column.ParquetProperties.newColumnWriteStore(ParquetProperties.java:222)
2024-03-06T21:03:24.6125853Z 	at org.apache.parquet.hadoop.InternalParquetRecordWriter.initStore(InternalParquetRecordWriter.java:116)
2024-03-06T21:03:24.6127582Z 	at org.apache.parquet.hadoop.InternalParquetRecordWriter.<init>(InternalParquetRecordWriter.java:101)
2024-03-06T21:03:24.6129102Z 	at org.apache.parquet.hadoop.ParquetRecordWriter.<init>(ParquetRecordWriter.java:152)
2024-03-06T21:03:24.6130581Z 	at org.apache.parquet.hadoop.ParquetOutputFormat.getRecordWriter(ParquetOutputFormat.java:505)
2024-03-06T21:03:24.6132154Z 	at org.apache.parquet.hadoop.ParquetOutputFormat.getRecordWriter(ParquetOutputFormat.java:432)
2024-03-06T21:03:24.6133753Z 	at org.apache.parquet.hadoop.ParquetOutputFormat.getRecordWriter(ParquetOutputFormat.java:427)
2024-03-06T21:03:24.6135536Z 	at org.apache.hadoop.hive.ql.io.parquet.write.ParquetRecordWriterWrapper.<init>(ParquetRecordWriterWrapper.java:70)
2024-03-06T21:03:24.6137725Z 	at org.apache.hadoop.hive.ql.io.parquet.MapredParquetOutputFormat.getParquerRecordWriterWrapper(MapredParquetOutputFormat.java:137)
2024-03-06T21:03:24.6140128Z 	at com.facebook.presto.hive.parquet.write.TestMapredParquetOutputFormat.getHiveRecordWriter(TestMapredParquetOutputFormat.java:61)
2024-03-06T21:03:24.6142089Z 	at com.facebook.presto.hive.parquet.ParquetTester.writeParquetColumn(ParquetTester.java:698)
2024-03-06T21:03:24.6143925Z 	at com.facebook.presto.hive.parquet.ParquetTester.assertRoundTrip(ParquetTester.java:363)
2024-03-06T21:03:24.6145444Z 	at com.facebook.presto.hive.parquet.ParquetTester.testRoundTrip(ParquetTester.java:307)
2024-03-06T21:03:24.6147095Z 	at com.facebook.presto.hive.parquet.ParquetTester.testRoundTrip(ParquetTester.java:245)
2024-03-06T21:03:24.6148945Z 	at com.facebook.presto.hive.parquet.AbstractTestParquetReader.testDecimalBackedByINT32(AbstractTestParquetReader.java:894)
2024-03-06T21:03:24.6150533Z 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2024-03-06T21:03:24.6151688Z 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2024-03-06T21:03:24.6153065Z 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2024-03-06T21:03:24.6154185Z 	at java.lang.reflect.Method.invoke(Method.java:498)
2024-03-06T21:03:24.6155438Z 	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:135)
2024-03-06T21:03:24.6156944Z 	at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:673)
2024-03-06T21:03:24.6158214Z 	at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:220)
2024-03-06T21:03:24.6159559Z 	at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
2024-03-06T21:03:24.6160997Z 	at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:945)
2024-03-06T21:03:24.6162441Z 	at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:193)
2024-03-06T21:03:24.6163831Z 	at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
2024-03-06T21:03:24.6165204Z 	at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
2024-03-06T21:03:24.6166452Z 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
2024-03-06T21:03:24.6167717Z 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
2024-03-06T21:03:24.6168669Z 	at java.lang.Thread.run(Thread.java:750)
2024-03-06T21:03:24.6169063Z 
2024-03-06T21:03:25.0839602Z [INFO] 
2024-03-06T21:03:25.0840062Z [INFO] Results:
2024-03-06T21:03:25.0840482Z [INFO] 
2024-03-06T21:03:25.0842587Z [ERROR] Failures: 
2024-03-06T21:03:25.0874877Z [ERROR]   TestFullParquetReader>AbstractTestParquetReader.testDecimalBackedByINT32:894 » IllegalArgument maxCapacityHint can't be less than initialSlabSize 1024 100
```

--- a/presto-hive/src/test/java/com/facebook/presto/hive/parquet/AbstractTestParquetReader.java
+++ b/presto-hive/src/test/java/com/facebook/presto/hive/parquet/AbstractTestParquetReader.java
@@ -130,8 +130,6 @@ public abstract class AbstractTestParquetReader
     private static final int MAX_PRECISION_INT32 = (int) maxPrecision(4);
     private static final int MAX_PRECISION_INT64 = (int) maxPrecision(8);
 
-    private Logger parquetLogger;
-
     private final ParquetTester tester;
 
     public AbstractTestParquetReader(ParquetTester tester)
@@ -145,7 +143,7 @@ public abstract class AbstractTestParquetReader
         assertEquals(DateTimeZone.getDefault(), HIVE_STORAGE_TIME_ZONE);
 
         // Parquet has excessive logging at INFO level
-        parquetLogger = Logger.getLogger("org.apache.parquet.hadoop");
+        Logger parquetLogger = Logger.getLogger("org.apache.parquet.hadoop");
         parquetLogger.setLevel(Level.WARNING);
     }
 
@@ -169,7 +167,7 @@ public abstract class AbstractTestParquetReader
     public void testNestedArrays()
             throws Exception
     {
-        int nestingLevel = ThreadLocalRandom.current().nextInt(1, 15);
+        int nestingLevel = 8;
         ObjectInspector objectInspector = getStandardListObjectInspector(javaIntObjectInspector);
         Type type = new ArrayType(INTEGER);
         Iterable values = limit(cycle(asList(1, null, 3, null, 5, null, 7, null, null, null, 11, null, 13)), 3_210);
@@ -186,7 +184,7 @@ public abstract class AbstractTestParquetReader
     public void testSingleLevelSchemaNestedArrays()
             throws Exception
     {
-        int nestingLevel = ThreadLocalRandom.current().nextInt(1, 15);
+        int nestingLevel = 5;
         ObjectInspector objectInspector = getStandardListObjectInspector(javaIntObjectInspector);
         Type type = new ArrayType(INTEGER);
         Iterable values = intsBetween(0, 31_234);
@@ -338,7 +336,7 @@ public abstract class AbstractTestParquetReader
     public void testNestedMaps()
             throws Exception
     {
-        int nestingLevel = ThreadLocalRandom.current().nextInt(1, 15);
+        int nestingLevel = 12;
         Iterable<Integer> keys = intsBetween(0, 3_210);
         Iterable maps = limit(cycle(asList(null, "value2", "value3", null, null, "value6", "value7")), 3_210);
         ObjectInspector objectInspector = getStandardMapObjectInspector(javaIntObjectInspector, javaStringObjectInspector);
@@ -555,7 +553,7 @@ public abstract class AbstractTestParquetReader
     public void testNestedStructs()
             throws Exception
     {
-        int nestingLevel = ThreadLocalRandom.current().nextInt(1, 15);
+        int nestingLevel = 14;
         Optional<List<String>> structFieldNames = Optional.of(singletonList("structField"));
         Iterable<?> values = limit(cycle(asList(1, null, 3, null, 5, null, 7, null, null, null, 11, null, 13)), 3_210);
         ObjectInspector objectInspector = getStandardStructObjectInspector(structFieldNames.get(), singletonList(javaIntObjectInspector));
@@ -883,8 +881,9 @@ public abstract class AbstractTestParquetReader
     public void testDecimalBackedByINT32()
             throws Exception
     {
+        int[] scales = {0, 0, 2, 0, 0, 2, 5, 5, 7, 4};
         for (int precision = 1; precision <= MAX_PRECISION_INT32; precision++) {
-            int scale = ThreadLocalRandom.current().nextInt(precision);
+            int scale = scales[precision];
             MessageType parquetSchema = parseMessageType(format("message hive_decimal { optional INT32 test (DECIMAL(%d, %d)); }", precision, scale));
             ContiguousSet<Integer> intValues = intsBetween(1, 1_000);
             ImmutableList.Builder<SqlDecimal> expectedValues = new ImmutableList.Builder<>();
@@ -926,8 +925,9 @@ public abstract class AbstractTestParquetReader
     public void testDecimalBackedByINT64()
             throws Exception
     {
+        int[] scales = {9, 2, 4, 1, 6, 3, 7, 1, 7, 12};
         for (int precision = MAX_PRECISION_INT32 + 1; precision <= MAX_PRECISION_INT64; precision++) {
-            int scale = ThreadLocalRandom.current().nextInt(precision);
+            int scale = scales[precision - MAX_PRECISION_INT32 - 1];
             MessageType parquetSchema = parseMessageType(format("message hive_decimal { optional INT64 test (DECIMAL(%d, %d)); }", precision, scale));
             ContiguousSet<Long> longValues = longsBetween(1, 1_000);
             ImmutableList.Builder<SqlDecimal> expectedValues = new ImmutableList.Builder<>();
@@ -942,8 +942,9 @@ public abstract class AbstractTestParquetReader
     public void testDecimalBackedByFixedLenByteArray()
             throws Exception
     {
+        int[] scales = {7, 13, 14, 8, 16, 20, 8, 4, 19, 25, 15, 23, 17, 2, 23, 0, 33, 8, 3, 12};
         for (int precision = MAX_PRECISION_INT64 + 1; precision < MAX_PRECISION; precision++) {
-            int scale = ThreadLocalRandom.current().nextInt(precision);
+            int scale = scales[precision - MAX_PRECISION_INT64 - 1];
             ContiguousSet<BigInteger> values = bigIntegersBetween(BigDecimal.valueOf(Math.pow(10, precision - 1)).toBigInteger(), BigDecimal.valueOf(Math.pow(10, precision)).toBigInteger());
             ImmutableList.Builder<SqlDecimal> expectedValues = new ImmutableList.Builder<>();
             ImmutableList.Builder<HiveDecimal> writeValues = new ImmutableList.Builder<>();

